pandas/core/indexes/datetimes.py

```diff
@@ -2087,6 +2094,8 @@ def _generate_regular_range(start, end, periods, freq):
                              "if a 'period' is given.")
 
         data = np.arange(b, e, stride, dtype=np.int64)
+
+        # _simple_new is getting an array of int64 here
         data = DatetimeIndex._simple_new(data, None, tz=tz)
```

Comment:
There is a new assert statement in ``_simple_new`` to check whether the input is an array of datetime64[ns]. But in this case data is an array of int64 so the assert statement fails. Is there a convenient way to rewrite this part to make data an array of datetime64[ns] before calling ``_simple_new`` so the assert works?

Reply:
oh yes, ``.view(_NS_DTYPE)``